### PR TITLE
📖 update releasing.md

### PR DESCRIPTION
**Pull request by tuminoid**
Update releasing.md with up to date steps for creating a release.

### CODE REVIEW

**Comment by tuminoid** [comment]
/test-ubuntu-integration-main
/test-centos-integration-main
/test-centos-e2e-integration-main

Using this PR to test dev-env as well.

**Comment by tuminoid** [comment]
CI should be stable now:

/test-centos-integration-main
/test-centos-e2e-integration-main

**Comment by tuminoid** [comment]
Iterating the document format and content at CAPM3 PR: https://github.com/metal3-io/cluster-api-provider-metal3/pull/916 then reflecting it here.

File: docs/releasing.md [hunk 1]
```diff
@@ -1,81 +1,114 @@
...
+  another Git tag with directory prefix, ie.
+  `git tag -s -a api/v1.x.y -m api/v1.x.y`.
 
-CAPM3 should use the latest version. Changes are required there to pull the
-latest version.
+### Release artifacts
 
-### Update Metal3-dev-env
+We need to verify all release artifacts are correctly built or generated by
+the release workflow. For a release, we should have the following artefacts:
```
Comment (smoshiur1237) [inline docs/releasing.md:72]:
```suggestion
the release workflow. For a release, we should have the following artifacts:
```

File: docs/releasing.md [hunk 2]
```diff
@@ -1,81 +1,114 @@
...
+  whatever release creation tagged to be manually checked.
+- For any superseded PRs (like CAPI uplifts, or commit revertions) that provide
+  no value to the release, create a summary line in `Superseded` section with
+  the PR ids and summary title. This way the changes are acknowledged to be part
+  of the release, but not overwhelming the important changes contained by the
+  release.
+- If the release you're making isn't the latest release (new
+  minor or major, or new patch release in the latest minor release),
+  uncheck the box for latest release.
+- If it is an release candidate (RC) or a pre-release, tick pre-release box.
```
[the remark this answers is not in the document]
Comment (tuminoid) [reply docs/releasing.md:101]: Fixed

File: docs/releasing.md (hunk 1 above)
Comment (tuminoid) [reply docs/releasing.md:72]: British vs US spelling... Well, fixed.

**Review by kashifest — APPROVED** [review]

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Comment by tuminoid** [comment]
Waiting for Jenkins to recover from Cleura outage.

**Pull request merged**